keras/backend/tensorflow_backend.py

```diff
@@ -432,6 +436,10 @@ def switch(condition, then_expression, else_expression):
     return tf.python.control_flow_ops.cond(condition,
                                            lambda: then_expression,
                                            lambda: else_expression)
+def broadcasting_select(condition, A, B):
```

Comment:
I find this difficult to understand and rather special-case. Couldn't we do without adding this to the backend? Also if you don't have the same function for Theano, then it doesn't belong in the backend. You could put it in the `rnn` scope, or get rid of it altogether.